src/bp/lang-server/service/index.ts

```diff
@@ -45,75 +39,179 @@ export default class LanguageService {
     return this._ready
   }
 
-  listModels(): AvailableModel[] {
-    return _.values(this._models)
+  listFastTextModels = (): AvailableModel[] => {
+    return _.values(this._models).map(m => m.fastTextModel)
   }
 
   private _discoverModels() {
     if (!fs.existsSync(this.langDir)) {
-      return []
+      return
     }
 
     const files = fs.readdirSync(this.langDir)
-    return files.forEach(f => {
+
+    const filesInfo: ModelFileInfo[] = []
+
+    _.forEach(files, file => {
```

Comment:
Just so you know this `_.forEach` would have been a perfect use case for `.filter` & `.map` composition .
const FT_REGEX =  /(\w+)\.(\w+)\.(\d+)\.bin/i
const BPE_REGX = /(\w+)\.(\w+)\.bpe\.model/i
const filesInfo = files
.map(f => ({ft: f.match(FT_REGEX), bpe: f.match(BPE_REGEX)})
.filter(data => data.ft && data.bpe
.map(data => (
  //return your mapped data here (what you currently push in the array)
))